comp/forwarder/defaultforwarder/forwarder.go

```diff
@@ -28,20 +31,32 @@ type provides struct {
 }
 
 func newForwarder(dep dependencies) provides {
-	return NewForwarder(dep.Config, dep.Log, dep.Params)
+	return NewForwarder(dep.Config, dep.Log, dep.Lc, true, dep.Params)
 }
 
 // NewForwarder returns a new forwarder component.
 //
 //nolint:revive
-func NewForwarder(config config.Component, log log.Component, params Params) provides {
+func NewForwarder(config config.Component, log log.Component, lc fx.Lifecycle, ignoreLifeCycleError bool, params Params) provides {
 	if params.UseNoopForwarder {
 		return provides{
 			Comp: NoopForwarder{},
 		}
 	}
+	forwarder := NewDefaultForwarder(config, log, params.Options)
+
+	lc.Append(fx.Hook{
+		OnStart: func(context.Context) error {
+			err := forwarder.Start()
+			if ignoreLifeCycleError {
```

Comment:
In which cases we want to ignore the life cycle error?

Reply:
`forwarder.Start()` error was ignored [before](https://github.com/DataDog/datadog-agent/pull/24009/commits/7619c28b75271c75476ddd3e193901d9a915518d#diff-c363b00939bb2af8b9f63471bd173091aa6335b59371d59a2c83514c3f5e698cL272) so I keep the same behavior.